File: Java/lab15_Command/task_3_3/src/Television.java
public class Television implements Device {

    private int volume;
    private int chanel;
    private boolean isOn;

    public Television() {
        this.isOn = false;
        this.volume = 50;
        this.chanel = 1;
    }

    @Override
    public void on() {
        if (isOn) {
            return;
        }
        System.out.println("Television is on");
        isOn = true;
    }

    @Override
    public void off() {
        if (!isOn) {
            return;
        }
        System.out.println("Television is off");
        isOn = false;
    }

    @Override
    public void volumeUp() {
        if (isOn) {
            return;
        }
        volume++;
        System.out.println("Volume = " + volume);
    }

    @Override
    public void volumeDown() {
        if (isOn) {
            return;
        }
        volume--;
        System.out.println("Volume = " + volume);
    }

    @Override
    public void nextChanel() {
        if (isOn) {
            return;
        }
        chanel++;
        System.out.println("Chanel = " + chanel);
    }

    @Override
    public void prevChanel() {
        if (isOn) {
            return;
        }
        chanel--;
        System.out.println("Chanel = " + chanel);
    }
}

File: Cpp/lab15_Command/task_3_3/src/television.cpp
#include "television.hpp"

#include <iostream>


Television::Television()
: volume(50), chanel(1), isOn(false) {}

Television::~Television() {}

void Television::on()
{
  if (this->isOn)
  {
    return;
  }

  std::cout << "Television is on" << std::endl;
  this->isOn = true;
}

void Television::off()
{
  if (!(this->isOn))
  {
    return;
  }

  std::cout << "Television is off" << std::endl;
  this->isOn = false;
}

void Television::volumeUp()
{
  if (this->isOn)
  {
    return;
  }

  ++(this->volume);
  std::cout << "Volume = " << this->volume << std::endl;
}

void Television::volumeDown()
{
  if (this->isOn && (this->volume > 0))
  {
    return;
  }
  --(this->volume);
  std::cout << "Volume = " << this->volume << std::endl;
}

void Television::nextChanel()
{
  if (this->isOn)
  {
    return;
  }

  ++(this->chanel);
  std::cout << "Chanel = " << this->chanel << std::endl;
}

void Television::prevChanel()
{
  if (this->isOn && (this->chanel > 0))
  {
    return;
  }

  --(this->chanel);
  std::cout << "Chanel = " << this->chanel << std::endl;
}
